### PR TITLE
Unit tests for the Error Reporting  via Logging API Unit Tests

### PR DESCRIPTION
**Pull request by iantalarico**

### CODE REVIEW

File: apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.Tests/ErrorReporting/ErrorReportingExceptionLoggerTest.cs [hunk 1]
```diff
@@ -111,7 +114,7 @@ private HttpContext CreateComplexContext()
 
             return context;
         }
-
+
```
Comment (jskeet) [inline apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.Tests/ErrorReporting/ErrorReportingExceptionLoggerTest.cs]: Remove whitespace?

File: apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.Tests/ErrorReporting/ErrorEventToLogEntryConsumerTest.cs [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+            Assert.Equal(3, logEntries.Count());
+            foreach (var entry in logEntries)
+            {
+                Assert.Equal(s_monitoredResource, entry.Resource);
+                Assert.Contains(_logName, entry.LogName);
+                Assert.Equal(LogSeverity.Error, entry.Severity);
+            }
+
+            var logOne = logEntries.Where(l => l.Timestamp.Equals(eventOne.EventTime)).Single();
```
Comment (jskeet) [inline apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.Tests/ErrorReporting/ErrorEventToLogEntryConsumerTest.cs]: Put the events in an array instead, and then just loop over them to check?

File: apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.Tests/ErrorReporting/ErrorEventToLogEntryConsumerTest.cs [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+        public void ConvertErrorEvents()
+        {
+            var startTime = DateTime.UtcNow;
+            var eventOne = new ReportedErrorEvent
+            {
+                Message = "Event one!",
+                EventTime = Timestamp.FromDateTime(startTime),
+                Context = new ErrorContext
+                {
+                    HttpRequest = new HttpRequestContext() { Method = "GET" },
```
Comment (evildour) [inline apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.Tests/ErrorReporting/ErrorEventToLogEntryConsumerTest.cs]: Nit: comma is not needed.

File: apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.Tests/ErrorReporting/ErrorEventToLogEntryConsumerTest.cs [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+                Context = new ErrorContext
+                {
+                    ReportLocation = new SourceLocation() { LineNumber = 10 }
+                }
+            };
+
+            var eventThree = new ReportedErrorEvent
+            {
+                Message = "Event three!",
+                EventTime = Timestamp.FromDateTime(startTime.AddSeconds(4)),
```
Comment (evildour) [inline apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.Tests/ErrorReporting/ErrorEventToLogEntryConsumerTest.cs]: Same here.

File: apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.Tests/ErrorReporting/ErrorReportingExtensionTest.cs [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+                },
+                
+            };
+
+            var errorStruct = errorEvent.ToStruct();
+            Assert.NotEmpty(errorStruct?.Fields);
+            Assert.Equal(message, errorStruct.Fields["message"].StringValue);
+
+            var contextStruct = errorStruct.Fields["context"].StructValue;
+            Assert.NotEmpty(contextStruct?.Fields);
```
Comment (evildour) [inline apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.Tests/ErrorReporting/ErrorReportingExtensionTest.cs]: Should this be checking any of the field values?

File: apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.Tests/ErrorReporting/ErrorReportingExceptionLoggerTest.cs (hunk 1 above)
Comment (iantalarico) [inline apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.Tests/ErrorReporting/ErrorReportingExceptionLoggerTest.cs]: Done

File: apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.Tests/ErrorReporting/ErrorEventToLogEntryConsumerTest.cs (hunk 3 above)
Comment (iantalarico) [inline apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.Tests/ErrorReporting/ErrorEventToLogEntryConsumerTest.cs]: Done

File: apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.Tests/ErrorReporting/ErrorEventToLogEntryConsumerTest.cs (hunk 4 above)
Comment (iantalarico) [inline apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.Tests/ErrorReporting/ErrorEventToLogEntryConsumerTest.cs]: Done

File: apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.Tests/ErrorReporting/ErrorReportingExtensionTest.cs (hunk 5 above)
Comment (iantalarico) [inline apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.Tests/ErrorReporting/ErrorReportingExtensionTest.cs]: Sorry I don't follow.  I check some fields below.

This test is really just a sanity check that the JSON conversions work.

File: apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.Tests/ErrorReporting/ErrorEventToLogEntryConsumerTest.cs (hunk 2 above)
Comment (iantalarico) [inline apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.Tests/ErrorReporting/ErrorEventToLogEntryConsumerTest.cs]: Moved to loop above.

**Pull request merged**